tasks/containers/TaskWorker.js

```diff
@@ -99,6 +100,24 @@ export default compose(
       ? '/' : `/tasks/${parentTaskPlan.id}`
     actions.router.push(nextRoute)
   }
+
+  function handleComplete (taskplan) {
+    const { parentTaskPlan } = taskPlan
+    const nextRoute = isNil(parentTaskPlan)
+      ? '/' : `/tasks/${parentTaskPlan.id}`
+
+    const taskWork = {
+      taskPlanId: taskPlan.id,
+      taskRecipeId: taskplan.taskRecipeId,
+      workerAgentId: currentAgent.id,
+      params: {
+        // contextAgentId: group.id
+      }
+    }
+
+    actions.taskWorks.create(taskWork)
```

Comment:
we're using [redux-observable](https://redux-observable.js.org/) in [feathers-action](https://github.com/ahdinosaur/feathers-action), so it's async but currently doesn't return a promise it [returns a cancellation token](https://github.com/ahdinosaur/feathers-action-react/blob/master/index.js#L51-L56).

the pattern currently is you subscribe to everything you need upfront, so that when things change you're automatically subscribed. in this case, we want to be subscribed to all task works referenced by the current task plan, so that when we create a new task work we'll be immediately notified. this subscription stuff is the `query` and `shouldQueryAgain` functions in any container using `connectFeathers`.

i figured out how we could return a promise though: if we implement a [ready action](https://github.com/ahdinosaur/feathers-action/commit/24180dc526952c87121a012e1a7b66495ccfd16d) for all methods, then we could fire off the action and resolve or reject a promise based on ready or error. but i haven't done this, i've only recently had the idea.